packages/data-context/src/data/ProjectLifecycleManager.ts

```diff
@@ -373,6 +377,19 @@ export class ProjectLifecycleManager {
 
     this.ctx.update((s) => {
       s.currentProject = projectRoot
+      s.currentProjectGitInfo?.destroy()
+      s.currentProjectGitInfo = new GitDataSource({
```

Comment:
I wonder if `DataSource` is the right suffix here, now we've changed it's lifecycle and where it's assigned? 

It feels like this is a little different to the other `Sources` we have now - all the others are created and cached on the `ctx.sources`, but not this one. What do you think?

Reply:
I don't think we need to change it right now, in that the `Source` is indicating it's "Reading" data, and `Action` indicates it's writing data. This is sort of the direction that we had discussed wanting to move, having clearer contracts for each class where we're injecting the minimum amount of state into the class for better isolation / testability.